### ISSUE TITLE
Double eclipsed texture not computed properly?

### ISSUE THREAD

**Issue by Atque** [10110111/CalcMySky#14]
When calculating the textures (using `sample.atmo,` as input), no errors occur. However, when displaying the textures in Stellarium, solar eclipses look too dark, as if the double scattering eclipsed texture is missing. This is with the eclipse quality set to 1. See the screenshot below.

My textures:
![stellarium-178](https://user-images.githubusercontent.com/37028368/231666934-fc71719b-b32c-49d9-bf11-040918c9a7f9.png)

Default textures:
![stellarium-179](https://user-images.githubusercontent.com/37028368/231667218-e8230dba-b201-4d88-831e-d61efc2477b6.png)

In ShowMySky GUI, however, my textures and the textures included in the CMS release look identical.

My setup is Windows 10 64-bit, AMD Radeon RX 5700 XT with driver 23.4.1.

**Comment by 10110111** [comment]
I've seen some problems with double scattering at eclipse on Linux/Intel UHD Graphics 620. It's strange that you are having similar problems on Windows with AMD.

> In ShowMySky GUI, however, my textures and the textures included in the CMS release look identical.

Please show the screenshots that you are comparing.

**Comment by Atque** [comment]
> I've seen some problems with double scattering at eclipse on Linux/Intel UHD Graphics 620. It's strange that you are having similar problems on Windows with AMD.

This card has previously shown signs of having troubles with CMS.

> Please show the screenshots that you are comparing.

My textures:
![result](https://user-images.githubusercontent.com/37028368/231838106-f2861da7-3f29-423a-ae03-887086d6e271.png)

Default textures:
![default](https://user-images.githubusercontent.com/37028368/231838184-1e88eb96-0532-49f1-827a-32790e9e23f2.png)

**Comment by 10110111** [comment]
Your screenshots are likely created with "Precompute double(-only) scattering on the fly" option turned on. This means the double scattering textures are ignored. What if you turn this option off?

**Comment by Atque** [comment]
Oops, you are correct. Disabling that shows the exact same difference in ShowMySky GUI.

I also notice that computing the textures on the fly in Stellarium (eclipse mode 2 and 3) also causes this problem.

**Comment by 10110111** [comment]
Does it work better if you set `radial integration points` value to a power of two like 64 instead of 50?

**Comment by 10110111** [comment]
Another problem on Intel/Mesa 21.2.6 is that with the default `iris` driver, uniforms passed to the shader stop getting updated at some point, getting stuck at one of the old values (even though `glGetUniform` returns the correct values). This affects at least `cameraViewDir`, while some uniforms have values that differ by several percent from the values passed. This is visible in the previewer with on-the-fly double scattering enabled, when the Moon is far from the Sun: the whole scene gets pale. Using `i965` driver via `MESA_LOADER_DRIVER_OVERRIDE=i965` makes this issue disappear.

So, in summary, on Ubuntu 20.04 Mesa 21.2.6 Intel UHD Graphics 620, I have to use a power-of-two `radial integration points` setting, and `i965` driver to make things work as expected.

**Comment by Atque** [comment]
> Does it work better if you set `radial integration points` value to a power of two like 64 instead of 50?

That seems to have solved this. At least the computed textures now look like the ones distributed in the release package. Thanks!

**Comment by 10110111** [comment]
Please check that this works as expected with the default value of 50 in 8c2809f981fc46a6df572418b6932f34c3cfab44 (not pushed to the master branch yet).
